impress-org/givewp
# Feature: add Forms tab to the campaign details page

Resolves [GIVE-1219]

## Description



This PR implements the Forms tab on the campaign details page where is possible to see the forms associated with the campaign and create new ones.

Please note that the list table here does not fully match the prototype design. As we are using the same list table component used in other places of the codebase, I think the best approach will be to change the list table design in a subsequent PR implementing the necessary changes and checking how it reflects in other views - like the subscriptions list.  

Also, the prototype design does not consider the pagination of the list table which can be a problem for campaigns with large amounts of forms, so probably a few discussions about it will be required.

## Affects



The "Forms" tab of the campaign details page.

## Visuals



![image](https://github.com/user-attachments/assets/9bf381ea-34e6-4bae-a59e-52930ec62b49)


## Testing Instructions



1. Go to `Give > Campaigns` page;
2. Create a new Campaign;
3. Go to the `Forms` tab and make sure a default form was created;
4. Create a new campaign form and make sure it gets visible in the `Forms` tab as well.

## Pre-review Checklist






[GIVE-1219]: https://stellarwp.atlassian.net/browse/GIVE-1219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
